proxy/coco/src/request/states.rs

```diff
@@ -0,0 +1,340 @@
+//! The state types and traits that we can use for [`super::Request`]'s `S` parameter.
+
+use std::{
+    collections::HashMap,
+    fmt,
+    ops::{Add, AddAssign, Deref},
+};
+
+use serde::{Deserialize, Serialize};
+
+use librad::{peer::PeerId, uri::RadUrn};
+
+use super::sealed;
+
+impl sealed::Sealed for Created {}
+impl sealed::Sealed for Requested {}
+impl sealed::Sealed for Found {}
+impl sealed::Sealed for Cloning {}
+impl sealed::Sealed for Cancelled {}
+
+// State Types
+
+/// An enumeration of the different states a `Request` can be in. This is useful if we want to
+/// convey the state information without any of the other state data.
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub enum RequestKind {
```

Comment:
```suggestion
pub enum RequestState {
``` 

would be truer to its meaning and purpose.

Reply:
Ya, I'm in agreement with this :+1: